### ISSUE TITLE
Training runs occasionally crash silently

### ISSUE THREAD

**Issue by tr1pzz** [NVlabs/stylegan2-ada#9]
Almost all of my training runs eventually (after 250k imgs or so) crash without an explicit error message...
This might have something to do with mixed precision training, but I'm really not sure:
![image](https://user-images.githubusercontent.com/24993604/96115427-36874100-0ee7-11eb-8d38-584d8deb3970.png)

**Comment by woctezuma** [comment]
For information, it does not happen to my training runs.

My first guess would be that you run out of GPU memory, so maybe tweak the config (mini-batch size, etc.).

**Comment by tr1pzz** [comment]
Yeah it might just be a GPU memory issue since I'm really close to max-memory when running 512x512 at batch_size of 8 on my RTX 2080 Ti, I'll look into it!

**Comment by tr1pzz** [comment]
My crashes are not due to insufficient memory, even when running with lower batch size where I'm only using 8/11 Gb train.py occasionally crashes silently...

**Comment by woctezuma** [comment]
The lack of error message is weird. Thankfully, I don't have this issue with my 256x256 JPG. I don't know what could be the root of the problem. I hope you update us about it when you get it solved.

**Comment by tr1pzz** [comment]
By now I'm actually pretty sure it's a memory leak issue as I can see the normal cache memory of my machine gradually filling up during a training run... Can anyone else confirm this?

**Comment by tr1pzz** [comment]
This is a memory profile generated using [mprof](https://github.com/pythonprofilers/memory_profiler)
![memory-profile](https://user-images.githubusercontent.com/24993604/97104613-87502400-16b5-11eb-80cb-fb0807813cd7.png)

**Comment by Foufix-** [comment]
I do now know if it's relevant, but I do encounter the same behavior (silent crash) everytime I try to do a training.
In my case, the Ram-OOM hypothesis has been confirmed by looking at the dmesg log. 
(GPU-OOM is much more verbose.)

**Comment by tr1pzz** [comment]
@tkarras @nurpax Do you guys have an idea what the source is of this leak? I'm sure you must have encountered this during your own training runs too?

**Comment by DELTA37** [comment]
I have inserted `gc.collect()` at the start of train cycle body (training_loop.py file) and `np.zeros(1 << 30)` => `np.zeros(1 << 25)` (labels creation dataset.py file) and problem have solved..

**Comment by nurpax** [comment]
@tr1pzz Alas, we don't really know.  We've had instances of what looked like a memory leak, but it doesn't happen in our normal training runs and we never root caused it further.

We're running on DGX-1 machines which have a lot of RAM -- this may have masked the problem for us.

**Comment by lukesionkirax** [comment]
Running into the same issue. Tried gc.collect() and np.zeros(1 << 30) => np.zeros(1 << 25) but still happening for me.

**Comment by tr1pzz** [comment]
Same here, @DELTA37's suggestion does not solve the issue for me...

**Comment by avillanueva** [comment]
I have the same problem and @DELTA37 suggestion does not help in my case. I have tried reducing batch sizes without success. In fact, I have the same problem with STYLEGAN2 when using configuration F, but it works with configuration E.

I have one question: is it possible to use config-e with STYLEGAN2-ADA?

**Comment by woctezuma** [comment]
> I have one question: is it possible to use config-e with STYLEGAN2-ADA?

Depending on your choice of parameters, you might already be using something else than config-e or config-f.

**Comment by dvschultz** [comment]
Just here to +1 this. I’m seeing similar things. I have a good amount of RAM (96GB) so it takes a while, but I still run into it eventually.

**Comment by taki0112** [comment]
sample problem +1

**Comment by johndpope** [comment]
This repo seems abandoned/ use pytorch 
https://github.com/NVlabs/stylegan2-ada-pytorch

**Comment by BrandoZhang** [comment]
> This repo seems abandoned/ use pytorch
> https://github.com/NVlabs/stylegan2-ada-pytorch

Memory leak still occurs in PyTorch version. I log `cpumem` onto tensorboard:
![image](https://user-images.githubusercontent.com/23352923/126634022-d0d88fa7-e920-4fd9-9b65-3dc935275027.png)

**Comment by waffletower** [comment]
I didn't see the memory leak when training with an RTX 2080 Ti.  Definitely seeing all sorts of weirdness, augmentation slowdowns, and definitely a repeatable CPU side memory leak when training using an RTX 3090.  Planning to test with the pytorch version.

**Comment by Ed-ivan** [comment]
> I have the same issue
